src/main/java/io/confluent/idesidecar/restapi/resources/OAuthCallbackResource.java

```diff
@@ -109,15 +114,40 @@ public Uni<String> callback(@QueryParam("code") String authorizationCode,
     }
   }
 
+  /**
+   * Returns the URI that the callback page should redirect to for a specific CCloud connection.
+   * Attempts to read the URI from the connection's spec. If null, returns the URI set via the
+   * application config <code>ide-sidecar.connections.ccloud.oauth.vscode-extension-uri</code>.
+   *
+   * @param connection the CCloud connection
+   * @return the URI
+   */
+  private String getRedirectUri(CCloudConnectionState connection) {
+    var connectionSpec = connection.getSpec();
+
+    if (connectionSpec != null) {
+      var ccloudConfig = connectionSpec.ccloudConfig();
+      if (ccloudConfig != null && ccloudConfig.ideAuthCallbackUri() != null) {
+        return ccloudConfig.ideAuthCallbackUri();
+      }
+    }
+
+    return CCLOUD_OAUTH_VSCODE_EXTENSION_URI;
```

Comment:
If the CCloud connection doesn't specify the `ide_auth_callback_uri`, we use the redirect URI set via the application config `ide-sidecar.connections.ccloud.oauth.vscode-extension-uri` by default.

@shouples I'd like to discuss this with you.